src/Uno.UI/UI/Xaml/UIElement.Layout.crossruntime.cs

```diff
@@ -304,13 +304,7 @@ public void Arrange(Rect finalRect)
 
 			var firstArrangeDone = IsFirstArrangeDone;
 
-			if (Visibility == Visibility.Collapsed
-				// If the layout is clipped, and the arranged size is empty, we can skip arranging children
-				// This scenario is particularly important for the Canvas which always sets its desired size
-				// zero, even after measuring its children.
-				|| (firstArrangeDone
-					&& finalRect == default
-					&& (this is not ICustomClippingElement clipElement || clipElement.AllowClippingToLayoutSlot)))
```

Comment:
This feels like we're changing an explicit behavior, were there additional information or tests around the time this was added?